kong.conf.default

```diff
@@ -1532,6 +1532,96 @@
                                  # same name/type will be synchronized to a
                                  # single query.
 
+#------------------------------------------------------------------------------
+# New DNS RESOLVER
+#------------------------------------------------------------------------------
+
+# This DNS resolver introduces global caching for DNS records across workers,
+# significantly reducing the query load on DNS servers.
+#
+# It provides observable statistics, you can retrieve them through the Admin API
+# `/status/dns`.
+
+#legacy_dns_client = off         # Disable the new DNS resolver, using the
+                                 # original DNS resolver. See above `dns_xxx`
+                                 # options for the original DNS resolver.
+
```

Comment:
hi @AndyZhang0707 do we need to expose this switch?

Reply:
Although we may use the legacy dns client as the default in the 3.8 release, we do want customers to try the new dns client thus I'd suggest to expose this switch in our doc.

btw, shall we name it as "use_legacy_dns_client"?